ggshield/cmd/utils/common_options.py

```diff
@@ -136,12 +136,32 @@ def log_file_callback(
 )
 
 
+def allow_self_signed_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[bool]
+) -> Optional[bool]:
+    if value:
+        ui.display_warning(
+            "The --allow-self-signed option is deprecated and will be removed in a future version. "
+            "Use --insecure instead, which more accurately describes what this option does "
+            "(it disables all certificate validation, not just allows self-signed certificates)."
```

Comment:
This is quite verbose. It produces a wall of text when used:

```
$ ggshield --allow-self-signed api-status
Warning: The --allow-self-signed option is deprecated and will be removed in a future version. Use --insecure instead, which more accurately describes what this option does (it disables all certificate validation, not just allows 
self-signed certificates).
Warning: SSL verification is disabled. Your connection to the GitGuardian API is NOT encrypted and is vulnerable to man-in-the-middle attacks. Traffic, including API keys and scan results, can be intercepted and modified.
Warning: To securely use self-signed certificates with Python >= 3.10, disable this option and install your certificate in your system's trust store. See: https://docs.gitguardian.com/ggshield-docs/configuration#allow_self_signed
API URL: https://dashboard.gitguardian.com
Status: healthy
App version: v2.315.0
Secrets engine version: 2.149.0

Instance source: default
API key source: user config
```

I'd rather have a shorter version, especially since this is followed by the --insecure warning.

```suggestion
            "Use --insecure instead."
```